index.html

```diff
@@ -9025,6 +9039,267 @@ <h3><dfn>Take Element Screenshot</dfn></h3>
+</ol>
+
+<p>A <a>String</a> is <dfn>equivalent to an empty string</dfn> if it
+has zero length after removing all <a>whitespace</a> characters.
+
+<p>When required to <dfn>parse as an integer</dfn> with
+argument <var>input</var> an implementation must:
+
+<ol>
+  <li>Let <var>stripped</var> be the result of stripping all leading
+  and trailing <a>whitespace</a> characters from <var>input</var>.
+
+  <li>If <var>stripped</var> has zero length, return <a>error</a> with
+  status <a>invalid argument</a>.
+
+  <li>If <var>stripped</var> contains any characters outside the
+  range <code>U+0030</code> - <code>U+0039</code> (i.e. 0 - 9)
+  inclusive, return <a>error</a> with status <a>invalid argument</a>.
+
+  <li>Let <var>output</var> be the result of calling <a>parseInt</a>
+  with string <var>stripped</var> and radix <code>10</code>.
+
+  <li>Return <a>success</a> with data <var>output</var>.
+</ol>
+
+<section>
+<h3><dfn>Print Page</dfn></h3>
+
+<table class="simple jsoncommand">
+ <tr>
+  <th>HTTP Method
+  <th>URI Template
+ </tr>
+ <tr>
+  <td>POST
+  <td>/session/{<var>session id</var>}/print
+ </tr>
+</table>
+
+<p>The <a>remote end steps</a> are:
+
+<ol>
+ <li><p>If the <a>current top-level browsing context</a> is <a>no longer open</a>,
+  return <a>error</a> with <a>error code</a> <a>no such window</a>.
+
+ <li><p><a>Handle any user prompts</a> and return its value if it is an <a>error</a>.
+
+ <li><p>Let <var>orientation</var> be the result of <a>getting a
+  property with default</a> named <code>orientation</code> and with
+  default "<code>portrait</code>" from the <var>parameters</var>
+  argument.
+
+ <li><p>If <var>orientation</var> is not a <a>String</a> or does not
+ have one of the values "<code>landscape</code>" or
+ "<code>portrait</code>", return <a>error</a> with <a>error
+ code</a> <a>invalid argument</a>.
+
+ <li><p>Let <var>scale</var> be the result of <a>getting a
+  property with default</a> named <code>scale</code> and with
+  default <code>1</code> from the <var>parameters</var> argument.
+
+ <li><p>If <var>scale</var> is not a <a>Number</a>, or is less
+ than <code>0.1</code> or greater than <code>2</code>
+ return <a>error</a> with <a>error code</a> <a>invalid argument</a>.
+
+ <li><p>Let <var>background</var> be the result of <a>getting a
+  property with default</a> named <code>background</code> and with
+  default <code>false</code> from the <var>parameters</var> argument.
+
+ <li><p>If <var>background</var> is not a <a>Boolean</a>
+ return <a>error</a> with <a>error code</a> <a>invalid argument</a>.
+
+ <li><p>Let <var>page</var> be the result of <a>getting a property
+  with default</a> named <code>page</code> and with a default of an
+  empty <a>Object</a> from the <var>parameters</var> argument.
+
+ <li><p>Let <var>pageWidth</var> be the result of <a>getting a
+  property with default</a> named <code>width</code> and with a
+  default of <code>8.5</code> from <var>page</var>.
```

Comment:
So this is equivalent to US letter size? Any chance we can have the equivalents in either millimetres or centimetres? It seems unfortunate to write a spec in the 21st century that relies on imperial measurements.

Reply:
It makes my skin crawl. But US letter size is defined in inches, and there's precedent for using inches, so I went with that. Happy to change this to use `cm` if there's a general feeling that it's acceptable.

Reply:
OK, I changed this to use `cm` on the basis that although we can't, in general, have nice things, there's a basic level of human decency which we have to maintain.